code/__HELPERS/game.dm

```diff
@@ -214,17 +214,29 @@
 /proc/get_hearers_in_view(R, atom/source)
 	// Returns a list of hearers in view(R) from source (ignoring luminosity). Used in saycode.
 	var/turf/T = get_turf(source)
-	var/list/hear = list()
+	. = list()
 
 	if(!T)
-		return hear
-
-	var/list/range = get_hear(R, T)
-	for(var/atom/movable/A in range)
-		hear |= recursive_hear_check(A)
-
-	return hear
+		return
 
+	var/list/processing_list = list()  
+	if (R == 0) // if the range is zero, we know exactly where to look for, we can skip view
+		processing_list += T.contents // We can shave off one iteration by assuming turfs cannot hear

```

Comment:
the walls have ears

Reply:
i hope you're joking, i was not able to find a /turf with flags_1 & HEAR_1

Reply:
I am joking.
"the walls have ears" was a phrase used in war time to encourage people to be careful about what they discussed, because anyone could be a spy or informant, as if the walls themselves had ears.

similar vain to 'loose lips sink ships'